src/app/core/services/transaction.service.ts

```
@@ -192,14 +193,19 @@ export class TransactionService {
     fileUploads$: Observable<FileObject[]>
   ): Observable<Partial<Transaction>> {
-    const upsertTxn$ = this.upsert(txn);
-    return forkJoin([fileUploads$, upsertTxn$]).pipe(
-      switchMap(([fileObjs, transaction]) => {
-        const fileIds = fileObjs.map((fileObj) => fileObj.id);
-        if (fileIds.length > 0) {
-          return this.expensesService.attachReceiptsToExpense(transaction.id, fileIds).pipe(map(() => transaction));
+    return fileUploads$.pipe(
+      switchMap((fileObjs) => {
+        // txn contains only source key when capturing receipt
+        if (txn.hasOwnProperty('source') && Object.keys(txn).length === 1) {
```

Biome

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)